pkg/channel/http/http_channel.go

```diff
@@ -365,46 +373,97 @@ func (h *Channel) invokeMethodV1(ctx context.Context, req *invokev1.InvokeMethod
-			w.WriteHeader(clientResp.StatusCode)
-			_, _ = io.Copy(w, clientResp.Body)
+			if sse {
+				callerResponseWriter := req.HTTPResponseWriter()
+
+				callerResponseWriter.Header().Set(headerContentType, mimeEventStream)
+				callerResponseWriter.Header().Set(headerCacheControl, cacheNoCache)
+				callerResponseWriter.Header().Set(headerConnection, connectionKeepAlive)
+
+				flusher, ok := callerResponseWriter.(http.Flusher)
+				if !ok {
+					http.Error(callerResponseWriter, "Streaming not supported", http.StatusInternalServerError)
+					return
+				}
+
+				buf := make([]byte, 1024)
+				for {
+					n, cErr := clientResp.Body.Read(buf)
+					if n > 0 {
+						if _, writeErr := callerResponseWriter.Write(buf[:n]); writeErr != nil {
+							return
+						}
+						flusher.Flush()
+					}
+					if cErr != nil {
+						if cErr != io.EOF {
+							clientErr = cErr
+						}
+						break
+					}
+				}
+			} else {
+				copyHeader(w.Header(), clientResp.Header)
+				w.WriteHeader(clientResp.StatusCode)
+				_, _ = io.Copy(w, clientResp.Body)
+			}
 		}
 		if clientErr != nil {
 			err = clientErr
 		}
 	}))
 	execPipeline.ServeHTTP(rw, channelReq)
-	resp := rw.Result() //nolint:bodyclose
 
 	elapsedMs := float64(time.Since(startRequest) / time.Millisecond)
 
 	var contentLength int64
-	if resp != nil {
-		if resp.Header != nil {
-			contentLength, _ = strconv.ParseInt(resp.Header.Get("content-length"), 10, 64)
-		}
-	}
 
 	if err != nil {
+		// content-length is omitted in http streaming scenarios
 		diag.DefaultHTTPMonitoring.ClientRequestCompleted(ctx, channelReq.Method, req.Message().GetMethod(), strconv.Itoa(http.StatusInternalServerError), contentLength, elapsedMs)
 		return nil, err
 	}
 
-	rsp, err := h.parseChannelResponse(resp)
-	if err != nil {
-		diag.DefaultHTTPMonitoring.ClientRequestCompleted(ctx, channelReq.Method, req.Message().GetMethod(), strconv.Itoa(http.StatusInternalServerError), contentLength, elapsedMs)
-		return nil, err
-	}
+	if sse {
+		return nil, nil
+	} else {
+		resp := rw.Result() //nolint:bodyclose
 
-	diag.DefaultHTTPMonitoring.ClientRequestCompleted(ctx, channelReq.Method, req.Message().GetMethod(), strconv.Itoa(int(rsp.Status().GetCode())), contentLength, elapsedMs)
+		if resp != nil {
+			if resp.Header != nil {
+				contentLength, _ = strconv.ParseInt(resp.Header.Get("content-length"), 10, 64)
+			}
+		}
 
-	return rsp, nil
+		rsp, err := h.parseChannelResponse(resp)
+		if err != nil {
+			diag.DefaultHTTPMonitoring.ClientRequestCompleted(ctx, channelReq.Method, req.Message().GetMethod(), strconv.Itoa(http.StatusInternalServerError), contentLength, elapsedMs)
+			return nil, err
+		}
+
+		diag.DefaultHTTPMonitoring.ClientRequestCompleted(ctx, channelReq.Method, req.Message().GetMethod(), strconv.Itoa(int(rsp.Status().GetCode())), contentLength, elapsedMs)
+
+		return rsp, nil
+	}
+}
+
+func isSSE(r *http.Request) bool {
+	accept := r.Header.Get("Accept")
+	return strings.HasPrefix(accept, "text/event-stream")
```

Comment:
looks like to https://github.com/dapr/dapr/pull/8832#discussion_r2165423313?

Reply:
It looks like there are omissions in other places as well, and it would be great if they could be added.